### PR TITLE
46 make robot drive given distance in centimeters

### PR DESCRIPTION
**Pull request by Greve2001**
I have rewritten our RPC protocol to take a Drive message, which has to be given a distance to drive.
I have also made supporting Go code for the robot to support the RPC call.

It needs testing. I will need this to be merged before i can start testing it.

### CODE REVIEW

File: robot/main.go [hunk 1]
```diff
@@ -31,6 +31,13 @@ type motorRequest struct {
 	motor   *ev3dev.TachoMotor
 }
 
+// Radius values are in centimeters and the wheelBaseRadius is measured from the inner sides of the wheels.
+var wheelRadius = 3.4
+var wheelBaseRadius = 8.75
+
+var wheelCircumference = 2 * wheelRadius * math.Pi
+var wheelBaseCircumference = 2 * wheelBaseRadius * math.Pi
+
```
Comment (MikaelFangel) [inline robot/main.go:40]: I think that this should be 'const' as they should never change during execution. Also maybe wrap it in a struct (:

File: robot/main.go [hunk 2]
```diff
@@ -155,16 +164,50 @@ func (s *motorServer) Rotate(_ context.Context, in *pBuff.RotateRequest) (*pBuff
...
+	// Fetch each motor
+	for i := 0; i < 2; i++ {
+		request := in.GetMotors()[i]
+
+		motor, err := getMotorHandle(request.GetMotorPort().String(), request.GetMotorType().String())
+		if err != nil {
+			return &pBuff.StatusReply{ReplyMessage: false}, err
+		}
+
+		// Construct Motor Requests
```
Comment (MikaelFangel) [inline robot/main.go:181]: Can't figure out which code this comments refers to.

File: robot/main.go [hunk 3]
```diff
@@ -155,16 +164,50 @@ func (s *motorServer) Rotate(_ context.Context, in *pBuff.RotateRequest) (*pBuff
...
 
-	wheelCircumference := 2 * wheelRadius * math.Pi
-	wheelBaseCircumference := 2 * wheelBaseRadius * math.Pi
+	return numberOfRotations
+}
 
-	// Calculate the distance move by rotating 1 degree on a wheel
-	rotationInCm := (wheelBaseCircumference * float64(degrees)) / 360.
-	return int(rotationInCm / (wheelCircumference / 360))
+// Hey
```
Comment (MikaelFangel) [inline robot/main.go:210]: Mistake?

File: robot/main.go [hunk 4]
```diff
@@ -155,16 +164,50 @@ func (s *motorServer) Rotate(_ context.Context, in *pBuff.RotateRequest) (*pBuff
...
 func convertRobotRotationToWheelRotations(degrees int32) int {
-	// Radius values are in centimeters and the wheelBaseRadius is measured from the inner sides of the wheels.
-	wheelRadius := 3.4
-	wheelBaseRadius := 8.75
+	rotationInCm := (wheelBaseCircumference * float64(degrees)) / 360.
+	numberOfRotations := int(rotationInCm / (wheelCircumference / 360))
 
-	wheelCircumference := 2 * wheelRadius * math.Pi
-	wheelBaseCircumference := 2 * wheelBaseRadius * math.Pi
+	return numberOfRotations
```
Comment (MikaelFangel) [inline robot/main.go:207]: Maybe skip making a variable and just write the calculation as a return (:

**Review by MikaelFangel — COMMENTED** [review]
Looks fine, only have some comments about stylistic choices. Also rotate might need to be validated as correct on the robot (:

File: robot/main.go (hunk 1 above)
Comment (Greve2001) [reply robot/main.go:40]: Fixed in next commit

File: robot/main.go (hunk 4 above)
Comment (Greve2001) [reply robot/main.go:207]: Fixed in next commit

File: robot/main.go (hunk 2 above)
Comment (Greve2001) [reply robot/main.go:181]: Fixed in next commit

**Review by MikaelFangel — APPROVED** [review]

**Pull request merged**